executor/chroot/chroot_executor.go

```diff
@@ -66,7 +82,7 @@ func (e *Executor) Run(f def.Formula, j def.Job, d string) def.JobResult {
 }
 
 // Execute a formula in a specified directory. MAY PANIC.
-func (e *Executor) Execute(f def.Formula, j def.Job, d string) def.JobResult {
+func (e *Executor) Execute(f def.Formula, j def.Job, d string, outS, errS io.WriteCloser) def.JobResult {
```

Comment:
This method sig is starting to get a liiiittle scary.  Would it make sense to swap out `j`, `outS`, and `errS` for handing in a channel that's expected to yield the ready job object?